### ISSUE TITLE
ANY Magisk version: stub.apk does not download full Magisk app anymore

### ISSUE THREAD

**Issue by steadfasterX** [topjohnwu/Magisk#8474]
Device: Google Pixel 7 Pro
Android version: 13
Magisk version name: **any** (tested with v27.0 and v28.0)
Magisk version code: 28001

might be related: #8445 

Problem:

After flashing Magisk and rebooting (three times to be sure) the Magisk app cannot be downloaded.
This is what happens after clicking "Upgrade to full Magisk to finish the setup. Download and install" :

```
10-15 19:54:14.138  1379  3856 D CoreBackPreview: Window{7473d97 u0 Downloading}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@f1c5569, mPriority=0}
10-15 19:54:14.139  1379  3856 D CoreBackPreview: Window{4f5d1d5 u0 com.topjohnwu.magisk/x.COMPONENT_PLACEHOLDER_2}: Setting back callback null
10-15 19:54:14.140  5018  5056 D OpenGLRenderer: endAllActiveAnimators on 0xb400003e2e16b300 (RippleDrawable) with handle 0xb400002c47b86a20
10-15 19:54:14.140  1379  3856 W InputManager-JNI: Input channel object '4f5d1d5 com.topjohnwu.magisk/x.COMPONENT_PLACEHOLDER_2 (client)' was disposed without first being removed with the input manager!
10-15 19:54:14.147  1379  1473 I input_focus: [Focus request 7473d97 Downloading,reason=UpdateInputWindows]
10-15 19:54:14.150  1379  1472 V WindowManager: Unknown focus tokens, dropping reportFocusChanged
10-15 19:54:14.166  5018  5078 D TrafficStats: tagSocket(97) with statsTag=0xffffffff, statsUid=-1
10-15 19:54:14.208  1379  1639 I input_focus: [Focus entering 7473d97 Downloading (server),reason=Window became focusable. Previous reason: NOT_VISIBLE]
10-15 19:54:14.397  5018  5078 E g       : java.io.FileNotFoundException: https://cdn.jsdelivr.net/gh/topjohnwu/magisk-files@28.0/app-release.apk
10-15 19:54:14.397  5018  5078 E g       : 	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:255)
10-15 19:54:14.397  5018  5078 E g       : 	at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getInputStream(DelegatingHttpsURLConnection.java:211)
10-15 19:54:14.397  5018  5078 E g       : 	at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:30)
10-15 19:54:14.397  5018  5078 E g       : 	at v.V(SourceFile:11)
10-15 19:54:14.397  5018  5078 E g       : 	at P.e(SourceFile:1)
10-15 19:54:14.397  5018  5078 E g       : 	at u.run(SourceFile:10)
10-15 19:54:14.397  5018  5078 E g       : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
10-15 19:54:14.397  5018  5078 E g       : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
10-15 19:54:14.397  5018  5078 E g       : 	at java.lang.Thread.run(Thread.java:1012)
10-15 19:54:14.398  1379  4366 I wm_finish_activity: [0,97912838,344,com.topjohnwu.magisk/x.COMPONENT_PLACEHOLDER_2,app-request]
```

indeed the file `https://cdn.jsdelivr.net/gh/topjohnwu/magisk-files@28.0/app-release.apk` cannot be found when trying to open that in a browser so this could be likely the reason.
The browser says: `Failed to fetch topjohnwu/magisk-files@28.0 from GitHub.`

EDIT:

I have installed Magisk v27.0 and even though it worked there before (before the release of v28.0) it stops working there as well now. 

**so the whole thing seems to affect everyone who just have the stub apk available and trying to get Magisk fully installed.**

Actually for the same reason as it (not suprisingly) tries to download the latest stable release always. so since the release of v28.0 it tries  `https://cdn.jsdelivr.net/gh/topjohnwu/magisk-files@28.0/app-release.apk` for ALL magisk versions I guess.

```
10-16 11:51:07.925  1381  1621 I am_compact: [2833,.ShannonImsService,anon,97484,44276,52396,2908,-5536,0,-5536,5536,48,2,23748,925,15,3061956,-5120]
10-16 11:51:08.538  1381  1633 I input_interaction: Interaction with: 5f08f17 com.topjohnwu.magisk/x.COMPONENT_PLACEHOLDER_2 (server), [Gesture Monitor] swipe-up (server), [Gesture Monitor] edge-swipe (server), PointerEventDispatcher0 (server), 
10-16 11:51:08.569  1641  1654 I native  : I0000 00:00:1729072268.569076    1654 tf_lite_classifier.cc:400] Event stream classified as kNone
10-16 11:51:08.731  1381  2285 D CoreBackPreview: Window{9aee82 u0 Downloading}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@d9746fc, mPriority=0}
10-16 11:51:08.732  1381  2285 D CoreBackPreview: Window{5f08f17 u0 com.topjohnwu.magisk/x.COMPONENT_PLACEHOLDER_2}: Setting back callback null
10-16 11:51:08.733  4096  4111 D OpenGLRenderer: endAllActiveAnimators on 0xb40000462a7f3c00 (RippleDrawable) with handle 0xb4000033e4e5c390
10-16 11:51:08.733  1381  2285 W InputManager-JNI: Input channel object '5f08f17 com.topjohnwu.magisk/x.COMPONENT_PLACEHOLDER_2 (client)' was disposed without first being removed with the input manager!
10-16 11:51:08.734  4096  4132 D TrafficStats: tagSocket(102) with statsTag=0xffffffff, statsUid=-1
10-16 11:51:08.742  1381  1465 I input_focus: [Focus request 9aee82 Downloading,reason=UpdateInputWindows]
10-16 11:51:08.745  1381  1464 V WindowManager: Unknown focus tokens, dropping reportFocusChanged
10-16 11:51:08.803  1381  1633 I input_focus: [Focus entering 9aee82 Downloading (server),reason=Window became focusable. Previous reason: NOT_VISIBLE]
10-16 11:51:08.921  4096  4132 E g       : java.io.FileNotFoundException: https://cdn.jsdelivr.net/gh/topjohnwu/magisk-files@28.0/app-release.apk
10-16 11:51:08.921  4096  4132 E g       : 	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:255)
10-16 11:51:08.921  4096  4132 E g       : 	at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getInputStream(DelegatingHttpsURLConnection.java:211)
10-16 11:51:08.921  4096  4132 E g       : 	at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:30)
10-16 11:51:08.921  4096  4132 E g       : 	at v.V(SourceFile:11)
10-16 11:51:08.921  4096  4132 E g       : 	at P.e(SourceFile:1)
10-16 11:51:08.921  4096  4132 E g       : 	at u.run(SourceFile:10)
10-16 11:51:08.921  4096  4132 E g       : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
10-16 11:51:08.921  4096  4132 E g       : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
10-16 11:51:08.921  4096  4132 E g       : 	at java.lang.Thread.run(Thread.java:1012)
10-16 11:51:08.922  1381  2285 I wm_finish_activity: [0,135122471,375,com.topjohnwu.magisk/x.COMPONENT_PLACEHOLDER_2,app-request]
```
`Failed to fetch topjohnwu/magisk-files@28.0 from GitHub.`

**Comment by steadfasterX** [comment]
_the whole thing seems to affect **everyone/every Magisk version** where just the stub apk is available and trying to get Magisk fully installed._

**Comment by topjohnwu** [comment]
Fixed in 455b13b83c4dde60511e43a634c880317b1ba5fc

**Comment by steadfasterX** [comment]
> Fixed in [455b13b](https://github.com/topjohnwu/Magisk/commit/455b13b83c4dde60511e43a634c880317b1ba5fc)

many thx @topjohnwu ! afaiu this requires a new build to take affect?
and it was in fact only affected v28.0 right? I might had a left-over from a previous test when testing v27.0 I guess..
